typing/typecore.ml

```diff
@@ -4143,6 +4177,38 @@ and type_expect_
           ty_expected_explained
       in
       { exp with exp_loc = loc }
+  | Pexp_apply
+      ({ pexp_desc = Pexp_extension({
+         txt = "extension.exclave" | "ocaml.exclave" | "exclave" as txt}, PStr []) },
+       [Nolabel, sbody]) ->
+      if (txt = "extension.exclave") && not (Language_extension.is_enabled Local) then
+          raise (Typetexp.Error (loc, Env.empty, Unsupported_extension Local));
+      begin
+        match expected_mode.position with
+        | RNontail ->
+          raise (Error (loc, env, Exclave_in_nontail_position))
+        | RTail (mode, _) ->
+          (* mode' is non-tail-of-region, because currently our language cannot construct
+           region in the tail of another region.
+            With the introduction of explicit region we will need something more clever *)
```

Comment:
I would probably just say `RNontail` rather than "non-tail-of-region". The second line has weird indentation. s/construct region/construct a region/. The second sentence doesn't really add anything.